### DISCUSSION TITLE
Is there a join type that can pushdown join filter to one side's scan?

### DISCUSSION THREAD

**Discussion by alexwilcoxson-rel** [apache/datafusion#7158] [Ideas]
I was looking through the join options' code and it didn't look like this was possible. e.g.

LHS table produces small number of rows. Collect these and provide them as a filter pushed down to RHS scan to reduce files read.

**Comment by alamb** [comment]
To the best of my knowledge this is not supported today by DataFusion -- I think this type of filter pushdown is called "sideways information passing" in database literature and is a typical technique to improve the performance for column stores. 

I think it has come up before in the context of improving join performance (for example, on the TPCH benchmarks -- cc @Dandandan  and @mingmwang )

@alexwilcoxson-rel  would you be willing to help file a feature request for this item? I can provide some additional commentary / pointers as well

**Reply by alexwilcoxson-rel** [reply]
thanks @alamb ! I also have a POC doing this that I can clean up and include. Will probably later today or early next week if I can get to it

**Comment by mjgarton** [comment]
Did a feature request for this even happen?  I have a need for something similar.

**Reply by alexwilcoxson-rel** [reply]
https://datafusion.apache.org/blog/2025/09/10/dynamic-filters/ specifically the [section about hash joins](https://datafusion.apache.org/blog/2025/09/10/dynamic-filters/#hash-join-dynamic-filters)

In my case I was hoping to use it with delta-rs. I think to take advantage though some work would need to happen on the delta-rs side to take advantage of the dynamic filters

**Reply by alamb** [reply]
Note that @adriangb and @rkrishn7 are actively working on improving the filtering support:

- https://github.com/apache/datafusion/pull/17529

**Reply by mjgarton** [reply]
Thank you for the responses @alexwilcoxson-rel and @alamb . datafusion is still mostly new to me, but your replies have got me started on learning more in the right areas.
